DashAI/back/preprocessor/base_converter.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from abc import ABCMeta, abstractmethod
+from typing import Final
+
+from datasets import DatasetDict
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from DashAI.back.config_object import ConfigObject
+
+
+class BaseConverter(ConfigObject, BaseEstimator, TransformerMixin, metaclass=ABCMeta):
+    """Base class for all converters"""
+
+    TYPE: Final[str] = "Converter"
+
+    @abstractmethod
+    def fit(self, dataset: DatasetDict) -> BaseConverter:
```

Comment:
I'm pretty sure this annotation doesn't work, since BaseConverter is not yet defined inside this scope (you are just defining it outside).

The solution to this is as follows:

```suggestion
    def fit(self, dataset: DatasetDict) -> "BaseConverter":
```